tests/e2e/kubetest2-kops/deployer/common.go

```diff
@@ -28,8 +36,108 @@ func (d *deployer) init() error {
 func (d *deployer) initialize() error {
 	if d.commonOptions.ShouldBuild() {
 		if err := d.verifyBuildFlags(); err != nil {
-			return fmt.Errorf("init failed to check build flags: %s", err)
+			return fmt.Errorf("init failed to check build flags: %v", err)
+		}
+	}
+	if d.commonOptions.ShouldUp() || d.commonOptions.ShouldDown() {
+		if err := d.verifyKopsFlags(); err != nil {
+			return fmt.Errorf("init failed to check kops flags: %v", err)
+		}
+	}
+	if d.commonOptions.ShouldUp() {
+		if err := d.verifyUpFlags(); err != nil {
+			return fmt.Errorf("init failed to check up flags: %v", err)
 		}
 	}
 	return nil
 }
+
+// verifyKopsFlags ensures common fields are set for kops commands
+func (d *deployer) verifyKopsFlags() error {
+	if d.ClusterName == "" {
+		name, err := defaultClusterName(d.CloudProvider)
+		if err != nil {
+			return err
+		}
+		klog.Info("Using cluster name ", d.ClusterName)
+		d.ClusterName = name
+	}
+
+	if d.KopsBinaryPath == "" {
+		if ws := os.Getenv("WORKSPACE"); ws != "" {
+			d.KopsBinaryPath = path.Join(ws, "kops")
+		} else {
+			return errors.New("missing required --kops-binary-path")
+		}
+	}
+	_, err := os.Stat(d.KopsBinaryPath)
+	if err != nil {
+		return err
+	}
+
+	switch d.CloudProvider {
+	case "aws":
+	default:
+		return errors.New("unsupported --cloud-provider value")
+	}
+
+	return nil
+}
+
+// env returns a list of environment variables passed to the kops binary
+func (d *deployer) env() []string {
+	vars := d.Env
+	vars = append(vars, []string{
+		fmt.Sprintf("PATH=%v", os.Getenv("PATH")),
+		fmt.Sprintf("HOME=%v", os.Getenv("HOME")),
+		fmt.Sprintf("KOPS_STATE_STORE=%v", stateStore(d.CloudProvider)),
+		fmt.Sprintf("KOPS_FEATURE_FLAGS=%v", d.featureFlags()),
+		"KOPS_RUN_TOO_NEW_VERSION=1",
+	}...)
+	if d.CloudProvider == "aws" {
+		vars = append(vars, fmt.Sprintf("AWS_SHARED_CREDENTIALS_FILE=%v", os.Getenv("AWS_SHARED_CREDENTIALS_FILE")))
+	}
+	return vars
+}
+
+// featureFlags returns the kops feature flags to set
+func (d *deployer) featureFlags() string {
+	return "+SpecOverrideFlag"
+}
+
+// defaultClusterName returns a kops cluster name to use when ClusterName is not set
+func defaultClusterName(cloudProvider string) (string, error) {
```

Comment:
This is copied from [test-infra/scenarios/kubernetes_e2e.py](https://github.com/kubernetes/test-infra/blob/3d086fadf368f1653399abc9377185f7055572aa/scenarios/kubernetes_e2e.py#L231-L251) with the following simplifications:
* Batch jobs (triggered by tide when merging a batch of PRs) included "batch" in the name but it was always hashed with the unique BUILD_ID, so it was easier to remove "batch" and just hash the BUILD_ID which is just as unique
* Presubmits used the PR number if the job was configured to tear down the previous cluster, but none of the kops jobs do that, so now the cluster names of all jobs will only depend on JOB_NAME and BUILD_ID